### PR TITLE
Refactoring and tests

### LINKED ISSUE #9: PR-2 The application is able to find the current location of the user

### LINKED ISSUE #10: US-2 As a user, I want to see a map of the Concordia campus, so that I can locate myself in the campus
- Provide a better map of the Concordia campus than the one on Google map
- Integrate it properly to the Google map

### PR DESCRIPTION
**Pull request by danseremet**
### Done
- Added tests for everything.
- Fixed Sonarqube coverage not showing up.
- Easy way to see coverage is to look at report of karma with live reload on ```ng test```.
- Harder way to see coverage is to re run ```npm run sonar``` everytime (slow).
- Refactored the code base so that it follows our architecture.
- Cleaned code to our standards.
- Abstracted shared stuff (imported everywhere).
- Abstracted core stuff (imported in root module).
- Added index.ts barrels for cleaner imports.
- Fixed wrong imports/dependencies on incorrect Geolocation.
- Closes #10 & Closes #9 

### Todo (next steps)
- Move Map stuff from Homepage to its own MapPage.
- Create new pages/components for other issues.

### Please Reviewers Review
- __Verify on mobile that your location is being used as the center of the map (by default it shows the map at Concordia unless you are on mobile).__

### CODE REVIEW

File: src/app/core/services/map.service.spec.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+            expect(locationServiceSpy.getGeoposition).toHaveBeenCalledTimes(1);
+        });
+    });
+
+    describe('tilesLoadedHandler()', () => {
+
+        // class MockElementRef extends ElementRef {
+        //     nativeElement = {};
+        // }
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.spec.ts]: can I remove this?

File: src/app/core/services/map.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        });
+
+        // it('should call getGeoposition from locationService and get nothing', () => {
+        //     const { mapService, locationServiceSpy, googleApisServiceSpy } = testServiceSetup();
+
+        //     const mapElement = new MockElementRef({});
+        //     mapService.loadMap(mapElement);
+
+        //     expect(locationServiceSpy.getGeoposition).toHaveBeenCalledTimes(1);
+        // });
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.spec.ts]: can i remove this?

File: src/app/core/services/map.service.spec.ts (hunk 1 above)
Comment (danseremet) [reply src/app/core/services/map.service.spec.ts]: sure I might have missed to delete some commented code elsewhere as well (it should be deleted if its commented)

File: src/app/core/services/map.service.spec.ts (hunk 2 above)
Comment (danseremet) [reply src/app/core/services/map.service.spec.ts]: yes

File: src/app/core/services/map.service.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        url: '../../../assets/icon/center_marker.png',
+        scaledSize: new google.maps.Size(30, 30), // scaled size
+    };
+
+    /**
+     * Given a map reference create a map 
+     * @param mapElement the reference to the html map
+     */
+    async loadMap(mapElement: ElementRef): Promise<google.maps.Map<Element>> {
+        const mapOptions: google.maps.MapOptions = {
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts]: Why is this const?

File: src/app/core/services/map.service.ts [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+import { LatLngLiteral } from '@google/maps';
+@Injectable()
+export class MapService {
+    constructor(
+        private locationService: LocationService,
+        private googleApis: GoogleApisService
+    ) { }
+
+    // marker: google.maps.Marker;
+    icon: google.maps.Icon = {
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts]: why was this changed from marker to icon?

File: src/app/core/services/map.service.ts (hunk 3 above)
Comment (danseremet) [reply src/app/core/services/map.service.ts]: reference doesnt change

File: src/app/core/services/map.service.ts (hunk 4 above)
Comment (danseremet) [reply src/app/core/services/map.service.ts]: it was not changed, I commented it since it is not used but it should in fact be removed since we don't store anything in the service (except icon which is a default thing part of loading a map)
